paris_rollins.py

```diff
@@ -70,18 +74,83 @@ def run_worker(log_file_root, log_time, remote_ip, remote_port,
       pass
   if not os.path.exists(log_file_dir):
     log_worker('cannot create %s' % log_file_dir)
-    return
+    return False
   try:
     log_file = open(log_file_name, 'w')
   except IOError:
     log_worker('cannot open log file %s' % log_file_name)
-    return
-  log_worker('traceroute to %s' % remote_ip)
+    return False
   try:
     returncode = subprocess.call(command, stdout=log_file)
     log_file.close()
     if returncode != 0:
       log_worker('%s returned %d' % (log_command, returncode))
+      return False
   except OSError:
-    log_worker('could not run %s' % log_ommand)
-  return
+    log_worker('could not run %s' % log_command)
+    return False
+  return True
+
+# Test if an IP address has been seen within the timeout period.
+class RecentIPAddressCache(object):
+
+  def __init__(self, cache_timeout):
+    self.cache_timeout = cache_timeout
+    self.address_cache = {}
+    self.address_cache_time_buckets = {}
+
+  # Expire all addresses up to 2 cache timeout periods ago.
+  def expire(self, now_bucket):
+    expire_buckets = [] 
+    for bucket in self.address_cache_time_buckets.keys():
+      if bucket < now_bucket - 1:
+        expire_buckets.append(bucket)
+    for bucket in expire_buckets:
+      for address in self.address_cache_time_buckets[bucket]:
+        del self.address_cache[address]
+      del self.address_cache_time_buckets[bucket]
+
+  # Returns true if an address seen without one timeout period.
+  def isrecent(self, address):
```

Comment:
Should be `isRecent` or `is_recent`, but I'm more worried that a function that is named like a check is not idempotent.  Checks shouldn't change state. I'd need to see this used in context, but I'd kind of want two functions: `isRecent(address)` and `cache(address, results)`, and only the second would update anything.